python/ngen_conf/src/ngen/config/init_config/utils.py

```diff
@@ -0,0 +1,100 @@
+import re
+from enum import Enum
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generic,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from pydantic.generics import GenericModel
+from typing_extensions import Self
+
+if TYPE_CHECKING:
+    from pydantic.typing import AbstractSetIntStr, DictStrAny, MappingIntStrAny
+
+L = TypeVar("L", bound=str)
+
+# matches a number that can contain an e or E and a unit contained in square brackets examples:
+# 2[m] -> group 0: 2 group 2: m
+# 2.0[m] -> group 0: 2.0 group 2: m
+# 4.05[] -> group 0: 4.05 group 2: ""
+# 2e5[m] -> group 0: 2 group 1 e5 group 2: m
+# 1.8e-05[m h-1] -> group 0: 1.8 group 1: e-05 group 2: m h-1
+_FLOAT_UNIT_PAIR_RE_PATTERN = r"([-]?\d*\.\d+|[-]?\d+)([Ee][-+]?\d+)?\[([^\]]*)\]"
```

Comment:
checked this out on https://regex101.com/ and it looks pretty solid

Reply:
Thanks for checking it!